nixos/modules/profiles/macos-builder.nix

```diff
@@ -93,7 +104,23 @@ in
         };
       });
 
-  system.stateVersion = "22.05";
+  system.stateVersion = lib.mkDefault (throw ''
+    The macOS linux builder should not need a stateVersion to be set, but a module
+    has accessed stateVersion nonetheless.
+    Please inspect the trace of the following command to figure out which module
+    has a dependency on stateVersion.
+
+      nix-instantiate --argstr system x86_64-darwin -A darwin.builder --show-trace
+
+    or
+
+      nix-instantiate --argstr system aarch64-darwin -A darwin.builder --show-trace
+
+    If this error occurred while evaluating the static part of the option
```

Comment:
This paragraph might not be necessary because the macOS builder disables documentation: https://github.com/NixOS/nixpkgs/blob/e52e5e0e6bfb34b2eaba08eda8c55e8b79cb4136/nixos/modules/profiles/macos-builder.nix#L27-L28